hooks/validators/deny_unsafe_apparmor.go

```diff
@@ -17,10 +17,51 @@ func (v DenyUnsafeAppArmor) Validate(ctx context.Context, pod *corev1.Pod) field
 	var errs field.ErrorList
 
 	for k, v := range pod.Annotations {
-		if strings.HasPrefix(k, corev1.AppArmorBetaContainerAnnotationKeyPrefix) &&
-			v != corev1.AppArmorBetaProfileRuntimeDefault {
+		if strings.HasPrefix(k, corev1.DeprecatedAppArmorBetaContainerAnnotationKeyPrefix) &&
+			v != corev1.DeprecatedAppArmorBetaProfileRuntimeDefault {
 			errs = append(errs, field.Forbidden(p.Key(k), fmt.Sprintf("%s is not an allowed AppArmor profile", v)))
 		}
 	}
+
+	p = field.NewPath("spec").Child("SecurityContext")
+	hasPodAppArmorProfile := pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.AppArmorProfile != nil
+	if hasPodAppArmorProfile {
+		isTypeUnconfined := pod.Spec.SecurityContext.AppArmorProfile.Type == corev1.AppArmorProfileTypeUnconfined
+		isTypeRuntimeDefault := pod.Spec.SecurityContext.AppArmorProfile.Type == corev1.AppArmorProfileTypeRuntimeDefault
+		isTypeLocalhost := pod.Spec.SecurityContext.AppArmorProfile.Type == corev1.AppArmorProfileTypeLocalhost
+		hasNotAllowedType := !(isTypeUnconfined || isTypeRuntimeDefault || isTypeLocalhost)
```

Comment:
Confined is not allowed. Please check PSS.
https://kubernetes.io/docs/concepts/security/pod-security-standards/

Reply:
Sorry, it was my miss understamd.  confused undefine and unconfined.